src/java/org/apache/cassandra/hints/HintsStore.java

```diff
@@ -203,14 +213,21 @@ private void deleteHints(Predicate<HintsDescriptor> predicate)
         {
             dispatchDequeue.removeAll(removeSet);
             corruptedFiles.removeAll(removeSet);
+            // if delete(descriptor) throws, hence
+            for (HintsDescriptor descriptor : removeSet)
+                hintsSizes.remove(descriptor.fileName());
         }
     }
 
     void delete(HintsDescriptor descriptor)
     {
         File hintsFile = descriptor.file(hintsDirectory);
         if (hintsFile.tryDelete())
+        {
+            // this method is also called standalone (without deleteHints)
```

Comment:
You are calling this twice on the normal path. Just leave the one call in this method, in `finally` if need be.